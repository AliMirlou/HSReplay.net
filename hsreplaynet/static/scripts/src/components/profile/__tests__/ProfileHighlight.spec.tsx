import React from "react";
import ProfileHighlight from "../ProfileHighlight";
import renderer from "react-test-renderer";

describe("ProfileHighlight", () => {
	test("does render with string highlight", () => {
		const component = renderer.create(
			<ProfileHighlight
				header="HEADER"
				subtitle="SUBTITLE"
				highlight="HIGHLIGHT"
			/>,
		);
		const tree = component.toJSON();
		expect(tree).toBeDefined();
		expect(tree.children[0].children[0].children[0].children[0]).toBe(
			"HIGHLIGHT",
		);
		expect(tree.children[1].children[0].children[0].children[0]).toBe(
			"HEADER",
		);
		expect(tree.children[1].children[0].children[1].children[0]).toBe(
			"SUBTITLE",
		);
		expect(tree).toMatchSnapshot();
	}),
		test("does render with component highlight", () => {
			const component = renderer.create(
				<ProfileHighlight
					header="HEADER"
					subtitle="SUBTITLE"
					highlight={<figure id="TEST_ID" />}
				/>,
			);
			const tree = component.toJSON();
			expect(tree).toBeDefined();
			expect(tree.children[0].children[0].children[0].type).toBe(
				"figure",
			);
			expect(tree.children[0].children[0].children[0].props.id).toBe(
				"TEST_ID",
			);
			expect(tree).toMatchSnapshot();
		});
});
