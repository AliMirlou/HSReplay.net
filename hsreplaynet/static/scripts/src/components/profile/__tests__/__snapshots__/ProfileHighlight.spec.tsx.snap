// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileHighlight does render with component highlight 1`] = `
<div
  className="profile-highlight"
>
  <div
    className="profile-highlight-content"
  >
    <div>
      <figure
        id="TEST_ID"
      />
    </div>
  </div>
  <div
    className="profile-highlight-text"
  >
    <div>
      <h4>
        HEADER
      </h4>
      <p>
        SUBTITLE
      </p>
    </div>
  </div>
</div>
`;

exports[`ProfileHighlight does render with string highlight 1`] = `
<div
  className="profile-highlight"
>
  <div
    className="profile-highlight-content"
  >
    <div>
      <p>
        HIGHLIGHT
      </p>
    </div>
  </div>
  <div
    className="profile-highlight-text"
  >
    <div>
      <h4>
        HEADER
      </h4>
      <p>
        SUBTITLE
      </p>
    </div>
  </div>
</div>
`;
